satellite/metainfo/metainfo_test.go

```diff
@@ -435,6 +439,70 @@ func TestCommitSegmentPointer(t *testing.T) {
 	})
 }
 
+func TestValueAttributeInfo(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 6, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		apiKey := planet.Uplinks[0].APIKey[planet.Satellites[0].ID()]
+		uplink := planet.Uplinks[0]
+		config := uplink.GetConfig(planet.Satellites[0])
+		metainfo, streams, err := config.GetMetainfo(ctx, uplink.Identity)
+		require.NoError(t, err)
+		redScheme := config.GetRedundancyScheme()
+		encScheme := config.GetEncryptionScheme()
+		_, err = metainfo.CreateBucket(ctx, "myBucket", &storj.Bucket{PathCipher: encScheme.Cipher})
+		require.NoError(t, err)
+
+		metainfoClient, err := planet.Uplinks[0].DialMetainfo(ctx, planet.Satellites[0], apiKey)
+		require.NoError(t, err)
+
+		keyInfo := pb.ValueAttributionRequest{
+			PartnerId: []byte("PartnerID"),
+			UserId:    []byte("userID"),
+			BucketId:  []byte("myBucketName"),
+		}
+
+		{
+			// bucket with no items
+			_, err = metainfoClient.ValueAttributeInfo(ctx, "myBucket", "", -1, string(keyInfo.PartnerId), string(keyInfo.UserId))
+			require.NoError(t, err)
+
+			// no bucket exists
+			_, err = metainfoClient.ValueAttributeInfo(ctx, "myBucket1", "", -1, string(keyInfo.PartnerId), string(keyInfo.UserId))
+			require.NoError(t, err)
+		}
+		{
+			createInfo := storj.CreateObject{
+				RedundancyScheme: redScheme,
+				EncryptionScheme: encScheme,
+			}
+			obj, err := metainfo.CreateObject(ctx, "myBucket", "path", &createInfo)
+			require.NoError(t, err)
+
+			reader := bytes.NewReader([]byte("one fish two fish red fish blue fish"))
+			err = uploadStream(ctx, streams, obj, reader)
```

Comment:
To upload something you need only something like this https://github.com/storj/storj/blob/f1641af802667edc594b27a927d38adb281db31e/internal/testplanet/uplink_test.go#L37